### DISCUSSION TITLE
A way to share logic between Dapps

### DISCUSSION THREAD

**Discussion by aleixmorgadas** [symbol-dapp/community#6] [Framework Ideas]
I'm thinking to create a shared interface between Dapps on top of `symbol-dapp` in way that other projects can reuse that logic.

Something like :
![symbol-platform](https://user-images.githubusercontent.com/49761379/116144810-c00c9c80-a6dc-11eb-8ce5-177148da8347.png)

symbol.directory creates a library that acts as a plug and play component in `symbol-dapp` so that other projects, in this example symbolsocial.network` can reuse without needing to re-implement the logic.

By doing so, people can reuse and compose Dapps that are the result of smaller Dapps. You can see this approach a copy of the [Unix philosophy](https://en.wikipedia.org/wiki/Unix_philosophy), the same but applied in Dapps.

- Write programs that do one thing and do it well.
- Write programs to work together.
- Write programs to handle text streams, because that is a universal interface.

**Comment by unbless** [comment]
Very abstract, more details needed. At the moment, everything is simple and clear, there is nothing to discuss here.

**Reply by aleixmorgadas** [reply]
Yep. It's still abstract. It's an idea I've in mind about the capability of `symbol-dapp` to be able to include this kind of features.

I will update when I have something more tangible.

**Comment by evias** [comment]
I opened a subsequent discussion under Dapps Ideas to open a discussion about *metadata* associated to dapps in the context of a Project Directory as illustrated in the above.

Great idea, @aleixmorgadas, looking forward to discuss more.
